addons/mail/static/src/models/attachment/attachment.js

```diff
@@ -235,6 +235,27 @@ function factory(dependencies) {
             return clear();
         }
 
+        /**
+         * @private
+         * @returns {string|undefined}
+         */
+        _computeFormattedSize() {
+            if (!this.size) {
+                return clear();
+            }
+            const units = ['%s bytes', '%s Kb', '%s Mb', '%s Gb'];
+            let value = this.size;
+            let i = 0;
+            while (parseInt(value, 10) >= 1000 && i < units.length) {
+                value /= 1000;
+                i++;
+            }
+            return _.str.sprintf(
+                this.env._t(units[i]),
```

Comment:
`this.env._t(X)` where is is an unevaluated string.
So you have to wrap string per cases (bytes, Kb, Mb, Gb).